### PR TITLE
Db doc

### PR DESCRIPTION
**Pull request by BarteldKlasens**
documentation of the database.

### CODE REVIEW

File: transmart-data/ddl/postgres/deapp/de_subject_microarray_data.sql [hunk 1]
```diff
@@ -23,3 +23,15 @@ CREATE TABLE de_subject_microarray_data (
 --
 ALTER TABLE ONLY de_subject_microarray_data
     ADD CONSTRAINT de_subject_microarray_data_pk PRIMARY KEY (assay_id, probeset_id);
+
+--
+-- add documentation
+--
+COMMENT ON TABLE deapp.de_subject_microarray_data IS 'Table holds microarray Highdim values';
+
+COMMENT ON COLUMN de_subject_microarray_data.trial_name IS 'Name of the triel. E.g. SHARED_HD_CONCEPTS_STUDY_C_PR';
```
Comment (gijskant) [inline transmart-data/ddl/postgres/deapp/de_subject_microarray_data.sql]: @forus do you know if this column is actually used? I hope not.

File: transmart-data/ddl/postgres/deapp/de_subject_sample_mapping.sql [hunk 2]
```diff
@@ -47,3 +47,15 @@ CREATE INDEX de_subject_smpl_mpng_idx2 ON de_subject_sample_mapping USING btree
...
+--
+-- add documentation
+--
+COMMENT ON TABLE deapp.de_subject_sample_mapping IS 'Table links patients to Highdim samples';
+
+COMMENT ON COLUMN de_subject_sample_mapping.patient_id IS 'The patient id linking the patient_dimension';
+COMMENT ON COLUMN de_subject_sample_mapping.subject_id IS 'links to subject_id in de_subject_microarray_data';
+COMMENT ON COLUMN de_subject_sample_mapping.concept_code IS 'The highdim concept code. E.g. CTHD:HD:EXPLUNG';
+COMMENT ON COLUMN de_subject_sample_mapping.assay_id IS 'Id that links to assay_id in highdim tables like de_subject_microarray_data and de_rnaseq_transcript_data';
+COMMENT ON COLUMN de_subject_sample_mapping.trial_name IS 'Name of the trial this sample is part of.';
```
Comment (gijskant) [inline transmart-data/ddl/postgres/deapp/de_subject_sample_mapping.sql]: @forus do you know if this column is actually used? I hope not.

File: transmart-data/ddl/postgres/deapp/de_subject_sample_mapping.sql [hunk 3]
```diff
@@ -47,3 +47,15 @@ CREATE INDEX de_subject_smpl_mpng_idx2 ON de_subject_sample_mapping USING btree
...
 
+--
+-- add documentation
+--
+COMMENT ON TABLE deapp.de_subject_sample_mapping IS 'Table links patients to Highdim samples';
+
+COMMENT ON COLUMN de_subject_sample_mapping.patient_id IS 'The patient id linking the patient_dimension';
+COMMENT ON COLUMN de_subject_sample_mapping.subject_id IS 'links to subject_id in de_subject_microarray_data';
+COMMENT ON COLUMN de_subject_sample_mapping.concept_code IS 'The highdim concept code. E.g. CTHD:HD:EXPLUNG';
+COMMENT ON COLUMN de_subject_sample_mapping.assay_id IS 'Id that links to assay_id in highdim tables like de_subject_microarray_data and de_rnaseq_transcript_data';
```
Comment (gijskant) [inline transmart-data/ddl/postgres/deapp/de_subject_sample_mapping.sql]: Suggestion:
Used as primary key of this table (although it is not an actual primary key and there is not even a proper index for this column). This key is references by high dimensional data tables, like de_subject_microarray_data and de_rnaseq_transcript_data.

File: transmart-data/ddl/postgres/deapp/de_subject_sample_mapping.sql [hunk 4]
```diff
@@ -47,3 +47,15 @@ CREATE INDEX de_subject_smpl_mpng_idx2 ON de_subject_sample_mapping USING btree
 --
 CREATE INDEX idx_de_subj_smpl_trial_ccode ON de_subject_sample_mapping USING btree (trial_name, concept_code);
 
+--
+-- add documentation
+--
+COMMENT ON TABLE deapp.de_subject_sample_mapping IS 'Table links patients to Highdim samples';
```
Comment (gijskant) [inline transmart-data/ddl/postgres/deapp/de_subject_sample_mapping.sql]: Suggestion:
Table to store information about assays, which are associated with high dimensional observations. Assays are typically associated with a patient. There can be many assays per patient, which may be distinguised by sample code, tissue type, platform, etc.

File: transmart-data/ddl/postgres/deapp/de_subject_sample_mapping.sql [hunk 5]
```diff
@@ -47,3 +47,15 @@ CREATE INDEX de_subject_smpl_mpng_idx2 ON de_subject_sample_mapping USING btree
...
+
+COMMENT ON COLUMN de_subject_sample_mapping.patient_id IS 'The patient id linking the patient_dimension';
```
Comment (gijskant) [inline transmart-data/ddl/postgres/deapp/de_subject_sample_mapping.sql]: Should not be empty, although it is nullable.

File: transmart-data/ddl/postgres/deapp/de_subject_sample_mapping.sql [hunk 6]
```diff
@@ -47,3 +47,15 @@ CREATE INDEX de_subject_smpl_mpng_idx2 ON de_subject_sample_mapping USING btree
...
+COMMENT ON COLUMN de_subject_sample_mapping.subject_id IS 'links to subject_id in de_subject_microarray_data';
+COMMENT ON COLUMN de_subject_sample_mapping.concept_code IS 'The highdim concept code. E.g. CTHD:HD:EXPLUNG';
```
Comment (gijskant) [inline transmart-data/ddl/postgres/deapp/de_subject_sample_mapping.sql]: Refers to concept_cd in concept_dimension.

File: transmart-data/ddl/postgres/deapp/de_subject_sample_mapping.sql [hunk 7]
```diff
@@ -47,3 +47,15 @@ CREATE INDEX de_subject_smpl_mpng_idx2 ON de_subject_sample_mapping USING btree
 --
 CREATE INDEX idx_de_subj_smpl_trial_ccode ON de_subject_sample_mapping USING btree (trial_name, concept_code);
 
+--
+-- add documentation
+--
+COMMENT ON TABLE deapp.de_subject_sample_mapping IS 'Table links patients to Highdim samples';
+
+COMMENT ON COLUMN de_subject_sample_mapping.patient_id IS 'The patient id linking the patient_dimension';
+COMMENT ON COLUMN de_subject_sample_mapping.subject_id IS 'links to subject_id in de_subject_microarray_data';
```
Comment (gijskant) [inline transmart-data/ddl/postgres/deapp/de_subject_sample_mapping.sql]: Not correct. Should be:
Corresponds to a part of the sourcesystem_cd column of patient_dimension. The patient_id column should be used for properly referencing patients.

File: transmart-data/ddl/postgres/deapp/de_subject_sample_mapping.sql [hunk 8]
```diff
@@ -47,3 +47,15 @@ CREATE INDEX de_subject_smpl_mpng_idx2 ON de_subject_sample_mapping USING btree
...
+COMMENT ON COLUMN de_subject_sample_mapping.concept_code IS 'The highdim concept code. E.g. CTHD:HD:EXPLUNG';
+COMMENT ON COLUMN de_subject_sample_mapping.assay_id IS 'Id that links to assay_id in highdim tables like de_subject_microarray_data and de_rnaseq_transcript_data';
+COMMENT ON COLUMN de_subject_sample_mapping.trial_name IS 'Name of the trial this sample is part of.';
+COMMENT ON COLUMN de_subject_sample_mapping.gpl_id IS 'id of the GPL platform for this sample. links to de_gpl_info table';
+COMMENT ON COLUMN de_subject_sample_mapping.sample_cd IS 'Code linking to the sample';
```
Comment (gijskant) [inline transmart-data/ddl/postgres/deapp/de_subject_sample_mapping.sql]: No link.
Code to distinguish different samples for the same patient.

**Comment by BarteldKlasens** [comment]
- Added schemaSpy generated model of the database.
- comments on the oracle ddl need to be tested

File: transmart-data/ddl/postgres/i2b2demodata/concept_counts.sql [hunk 9]
```diff
@@ -12,4 +12,11 @@ CREATE TABLE concept_counts (
 --
 CREATE INDEX idx_count_concept_path ON concept_counts USING btree (concept_path);
 
+--
+-- add documentation
+--
+COMMENT ON TABLE i2b2demodata.concept_counts IS 'Stores number of patients for which there are observations over the given concept.';
```
Comment (gijskant) [inline transmart-data/ddl/postgres/i2b2demodata/concept_counts.sql]: Unused/deprecated?

File: transmart-data/ddl/postgres/i2b2demodata/qt_analysis_plugin.sql [hunk 10]
```diff
@@ -30,3 +30,8 @@ ALTER TABLE ONLY qt_analysis_plugin
 --
 CREATE INDEX qt_apnamevergrp_idx ON qt_analysis_plugin USING btree (plugin_name, version_cd, group_id);
 
+--
+-- add documentation
+--
+COMMENT ON TABLE i2b2demodata.observation_fact IS 'Not used by TranSMART. The plug-in’s metadata for an individual project is stored in this
```
Comment (gijskant) [inline transmart-data/ddl/postgres/i2b2demodata/qt_analysis_plugin.sql]: Comment is on wrong table. That one is actually used by transmart...
Maybe remove this comment entirely?

File: transmart-data/ddl/postgres/i2b2demodata/concept_counts.sql (hunk 9 above)
Comment (forus) [inline transmart-data/ddl/postgres/i2b2demodata/concept_counts.sql]: I'll add a note about it.

File: transmart-data/ddl/postgres/i2b2demodata/qt_analysis_plugin.sql (hunk 10 above)
Comment (forus) [inline transmart-data/ddl/postgres/i2b2demodata/qt_analysis_plugin.sql]: Yes, good idea.

**Pull request merged**